src/components/download-print-menu/download-print-menu.tsx

```diff
@@ -0,0 +1,171 @@
+    };
+    state: DownloadPrintMenuState = {
+        popoverOpen: false
+    };
+    private _controlElement: any;
+
+    componentDidMount() {
+        document.addEventListener("click", this._handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener("click", this._handleClickOutside);
+    }
+
+    // TODO: could be used for popover accessibility
+    // componentDidUpdate(prevProps: DownloadPrintMenuProps, prevState: DownloadPrintMenuState) {
+    //     if (!prevState.popoverOpen && this.state.popoverOpen) {
+    //         this._controlElement.children[0].focus();
+    //     }
+    // }
+
+    private _handleClickOutside = (e: any) => {
+        if (
+            !!this._controlElement &&
+            !this._controlElement.contains(e.target) &&
+            this.state.popoverOpen
+        ) {
+            this._handleClosePopover();
+        }
+    };
+
+    private _handleClosePopover = () => {
+        this.setState({ popoverOpen: false });
+    };
+
+    private _onButtonClick = () => {
+        this.setState((state: DownloadPrintMenuState) => ({ popoverOpen: !state.popoverOpen }));
+    };
+
+    private _onDownloadClicked = () => {
+        this._handleClosePopover();
+        this.props.onDownload();
+    };
+
+    private _onPrintClicked = () => {
+        this._handleClosePopover();
+        this.props.onPrint();
+    };
+    private _onKeyDown = (e: KeyboardEvent, cb: Function) => {
+        e.stopPropagation();
+        switch (e.keyCode) {
+            case 13: // Enter pressed
+                if (typeof cb === "function") {
+                    cb();
+                }
+                break;
+            case 27: // ESC pressed
+                this._handleClosePopover();
+                break;
+        }
+    };
+    private _popoverMenuItemRenderer = (el: PopoverMenuItem) => (
+        <div
+            tabIndex={1}
+            role="button"
+            onClick={() => el.onMenuChosen()}
+            onKeyDown={e => this._onKeyDown(e, el.onMenuChosen)}
+            className={styles.popoverMenuItem}
+        >
+            {el.label}
+        </div>
+    );
+
+    private _getPopoverMenuOptions = () => {
+        return [
+            {
+                label: this.props.downloadButtonAriaLabel,
+                onMenuChosen: this._onDownloadClicked
+            },
+            {
+                label: this.props.printButtonAriaLabel,
+                onMenuChosen: this._onPrintClicked
+            }
+        ];
+    };
+
+    private _renderButton = () => (
+        <button
+            tabIndex={1}
+            aria-label={this.props.dropdownAriaLabel}
+            className={styles.downloadPrintButton}
+        >
+            <div className={[styles.icon, styles.downloadIcon].join(" ")} />
+        </button>
+    );
+
+    render(props: DownloadPrintMenuProps) {
+        if (!props.downloadDisabled && !props.printDisabled) {
```

Comment:
Just a syntax thing - I am OK with leaving this as it is now. 
Personally I think that we can extract relevant props like `const {downloadDisabled, printDisabled} = props;`
It makes the rest of the render function a bit cleaner

Reply:
done